src/parser.rs

```diff
@@ -580,6 +580,10 @@ impl<'a> DocParser<'a> {
     let mut handled_symbols = HashSet::new();
 
     for (export_name, export_symbol_id) in symbol.exports() {
+      if export_name == &namespace_name
+      {
+        continue;
+      }
```

Comment:
What happens for the following code?

```ts
export declare namespace test {
    const test: string;
    export { test };
}
```

Or the following:

```ts
export declare namespace test {
    export class test {}
}
```

I think it can't be based on strings, but probably needs to be based on the symbol id or swc id.

Reply:
I previously attempted to use the symbol ID, but the output varied for certain inputs. For instance:

```
export function describe(name?: string, options?: TestOptions, fn?: SuiteFn): Promise<void>;
export function describe(name?: string, fn?: SuiteFn): Promise<void>;
export function describe(options?: TestOptions, fn?: SuiteFn): Promise<void>;
export function describe(fn?: SuiteFn): Promise<void>;

export namespace describe {
    /**
     * Shorthand for skipping a suite, equivalent to `describe([name], { skip: true }[, fn])`.
     */
    export function skip(name?: string, options?: TestOptions, fn?: SuiteFn): Promise<void>;
    export function skip(name?: string, fn?: SuiteFn): Promise<void>;
    export function skip(options?: TestOptions, fn?: SuiteFn): Promise<void>;
    export function skip(fn?: SuiteFn): Promise<void>;

    /**
     * Shorthand for marking a suite as `TODO`, equivalent to `describe([name], { todo: true }[, fn])`.
     */
    export function todo(name?: string, options?: TestOptions, fn?: SuiteFn): Promise<void>;
    export function todo(name?: string, fn?: SuiteFn): Promise<void>;
    export function todo(options?: TestOptions, fn?: SuiteFn): Promise<void>;
    export function todo(fn?: SuiteFn): Promise<void>;

    /**
     * Shorthand for marking a suite as `only`, equivalent to `describe([name], { only: true }[, fn])`.
     * @since v18.15.0
     */
    export function only(name?: string, options?: TestOptions, fn?: SuiteFn): Promise<void>;
    export function only(name?: string, fn?: SuiteFn): Promise<void>;
    export function only(options?: TestOptions, fn?: SuiteFn): Promise<void>;
    export function only(fn?: SuiteFn): Promise<void>;
}
```
The documentation is only generated for the first line: export function describe(name?: string, options?: TestOptions, fn?: SuiteFn): Promise<void>; The rest are skipped because they share the same symbol ID.

Reply:
I tested the two examples you mentioned, and encountered a panic on the debug assertion line:
![image](https://github.com/denoland/deno_doc/assets/168719085/2e7bed05-7a99-4436-9463-86f0360bdc8d)
After my changes, the function returns Some instead of None:
`self.visibility.get_root_exported_deps(&child_symbol.unique_id()).clone() = Some(
    SymbolDeps(
        {},
    ),
)`
From my understanding I think it is normal in this case to return some and I believe there is no issue with removing the debug assertion code that caused the panic. After removing the debug assertion line and retesting the two examples, everything worked correctly, generating the right documentation as shown in the following images:
 
![image](https://github.com/denoland/deno_doc/assets/168719085/1bbe9095-1a1d-4e67-8cc2-5dd530f1aefa)
![image](https://github.com/denoland/deno_doc/assets/168719085/faa413a9-1ce1-4f03-83cf-2b875efb14ec)

Reply:
Hello @dsherret 
I wanted to follow up on my previous comment. Could you please check my comment above when you get a chance?